batch/test/test_batch.py

```diff
@@ -604,6 +604,12 @@ def assert_job_ids(expected, q=None):
         j_failure.wait()
         j_error.wait()
 
+        delay = 0.1
+        while True:
+            if j_running.status()['state'] == 'Running' or j_running.is_complete():
+                break
+            delay = sync_sleep_and_backoff(delay)
```

Comment:
Can we pull this logic out into a function in the test utils? I see similar logic in `test_running_job_log_and_status`. That one should actually get updated to use this logic since it's a hot loop that probably causes a spike in requests.